### DISCUSSION TITLE
Client Routing. Behaviour while clicking back button in browser.

### DISCUSSION THREAD

**Discussion by vzhlznva** [vikejs/vike#677] [Discussion]
**EDITED**: 
I **fixed** this error with fixing error with page data object. But also I can't understand why when I'm clicking back button in browser, my page doesn't change. Also there is no request for `index.pageContext.json` 

**FIXED**
Hi there! I ran into a problem with client routing. 
When I go to a new page, I get this error in the console that stops routing, which makes me unable to go to the page:
![image](https://user-images.githubusercontent.com/56662385/222522145-98e3fa7a-5e46-46f1-9934-c7952829a70a.png)
![image](https://user-images.githubusercontent.com/56662385/222522232-653809f6-57a5-42f6-8d6b-137ecbcd9389.png)

Which is caused by this in file `_default.page.client.ts`:

`export const render = async (pageContext: PageContextBuiltInClient & PageContext) => {`
  `if (!app) {`
    `const instance = createApp(pageContext)`
    `app = instance.app`
    `instance.store.state.value = pageContext.initialStoreState`
    `app.mount('#app')`
 ` } else {`
    `app.changePage(pageContext)` <--
  `}`
  `document.title = getPageTitle(pageContext); `
`}`

and this in file `app.ts`: 

`objectAssign(app, {`
    `changePage: (pageContext: PageContext) => {`
      `console.log('change', pageContext)`
      `Object.assign(pageContextReactive, pageContext)`
      `rootComponent.Page = markRaw(pageContext.Page)` <--
      `rootComponent.pageProps = markRaw(pageContext.pageProps || {})`
    `}`
  `});`

Also I use Pinia to store fetched data

Most of the code is taken from the examples in your repository. 
I would be grateful for your help!

**Comment by lorrenrules** [comment]
I'm also experiencing difficulties implementing the `changePage` behaviors in a hybrid SSR/SPA app as detailed in the examples. The `changePage` method being added to the instance doesn't seem to work as it has in the past. When the root component is created and sets the outside variable `rootComponent` to itself, that context seems to be limited to the component now, as the value shows its initial value when used outside the component's context.

Thanks for the help.

**Reply by vzhlznva** [reply]
Have you found the way how to fix it? Or seen solutions? 
Because I really don't know how to fix it

**Reply by brillout** [reply]
> doesn't seem to work as it has in the past

I'd create a reproduction showcasing the regression and tell the Vue team about it. They'll probably give useful insights in the process.

Let me know what you ended up doing.

**Reply by lorrenrules** [reply]
I pulled down the `vue-full` example from the repo, and it works just fine. I'm not sure what's going on in my project, but I think it's clearly user error. If I do find reason to think it's a regression, I'll make a reproduction and post a new discussion post here or with the Vue team if it seems unrelated to VPS. Thanks for the response, and for your work on this project. Cheers.

**Comment by brillout** [comment]
It's probably unrelated to your issues, but FYI there was a bug with VPS's Client Routing behavior going back to a 404 page, which is now fixed https://github.com/brillout/vite-plugin-ssr/issues/681.
